test/pkg/ptptesthelper/ptptesthelper.go

```diff
@@ -272,53 +272,149 @@ func CheckSlaveSyncWithMaster(fullConfig testconfig.TestConfig) {
+	logrus.Info("Slave interfaces are ", slaveIf)
+	// Toggle the interfaces
+	for _, ptpNodeInterface := range slaveIf {
+		_, skip := skippedInterfaces[ptpNodeInterface]
+		if skip {
+			logrus.Info("Skipping the interface ", ptpNodeInterface)
+		} else {
+			logrus.Info("Simulating network outage using interface ", ptpNodeInterface)
+			toggleNetworkInterface(networkOutageRecoveryDaemonsetPod, ptpNodeInterface, slavePodNodeName, fullConfig)
+		}
 	}
+	DeletePtpTestPrivilegedDaemonSet(pkg.RecoveryNetworkOutageDaemonSetName, pkg.RecoveryNetworkOutageDaemonSetNamespace)
+	logrus.Info("Recovery network outage ends ...........")
+}
 
-	// 2. Get the slave config, restart the slave node
-	slavePtpConfig := fullConfig.DiscoveredClockUnderTestPtpConfig
-	restartedNodes := make([]string, len(rebootDaemonSetRunningPods.Items))
+func toggleNetworkInterface(pod corev1.Pod, interfaceName string, slavePodNodeName string, fullConfig testconfig.TestConfig) {
 
-	for _, recommend := range slavePtpConfig.Spec.Recommend {
-		matches := recommend.Match
+	const (
+		waitingPeriod = 1 * time.Minute
+	)
 
-		for _, match := range matches {
-			nodeLabel := *match.NodeLabel
-			// Get all nodes with this node label
-			nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: nodeLabel})
-			Expect(err).NotTo(HaveOccurred())
+	downInterfaceCommand := fmt.Sprintf("ip link set dev %s down", interfaceName)
+	logrus.Info("Setting the interface ", interfaceName, " down")
+	pods.ExecutePtpInterfaceCommand(pod, interfaceName, downInterfaceCommand)
+	logrus.Info("Interface ", interfaceName, " is set down")
 
-			// Restart the node
-			for _, node := range nodes.Items {
-				nodeshelper.RebootNode(nodeToPodMapping[node.Name], node)
-				restartedNodes = append(restartedNodes, node.Name)
-			}
-		}
+	time.Sleep(waitingPeriod)
+
+	// Check if the port state has changed to faulty
+	err := metrics.CheckClockRole(metrics.MetricRoleFaulty, interfaceName, &slavePodNodeName)
+	Expect(err).NotTo(HaveOccurred())
+
+	/* Check if the clock state is in sync
+	err = metrics.CheckClockState(metrics.MetricClockStateLocked, interfaceName, &slavePodNodeName)
+	Expect(err).To(HaveOccurred())
+	logrus.Info(err)*/
+
+	upInterfaceCommand := fmt.Sprintf("ip link set dev %s up", interfaceName)
+	pods.ExecutePtpInterfaceCommand(pod, interfaceName, upInterfaceCommand)
+	logrus.Info("Interface ", interfaceName, " is up")
+	time.Sleep(waitingPeriod)
+
+	// Check if the port has the role of the slave
+	err = metrics.CheckClockRole(metrics.MetricRoleSlave, interfaceName, &slavePodNodeName)
+	Expect(err).NotTo(HaveOccurred())
+	/* Check if the clock state is in LOCKED state
+	err = metrics.CheckClockState(metrics.MetricClockStateFreeRun, interfaceName, &slavePodNodeName)
+	logrus.Info(err)
+	Expect(err).NotTo(HaveOccurred())*/
+
+	offsetVal, err := metrics.GetPtpOffeset(interfaceName, &slavePodNodeName)
```

Comment:
Add a retry for 5 times to see offset gets withing boundary

Reply:
Done.